python/cuml/test/test_linear_model.py

```diff
@@ -488,3 +488,82 @@ def test_logistic_predict_convert_dtype(train_dtype, test_dtype):
     clf = cuLog()
     clf.fit(X_train, y_train)
     clf.predict(X_test.astype(test_dtype))
+
+
+@pytest.fixture(scope='session',
+                params=['binary', 'multiclass'])
+def regression_dataset(request):
+    regression_type = request.param
+    n_samples = 500000
+    n_features = 5
+
+    data = (np.random.rand(n_samples, n_features) * 2) - 1
+
+    if regression_type == 'binary':
+        coef = (np.random.rand(n_features) * 2) - 1
+        coef /= np.linalg.norm(coef)
+        output = (data @ coef) > 0
+    elif regression_type == 'multiclass':
+        n_classes = 3
+        coef = (np.random.rand(n_features, n_classes) * 2) - 1
+        coef /= np.linalg.norm(coef, axis=0)
+        output = (data @ coef).argmax(axis=1)
+
+    output = output.astype(np.int32)
+    return regression_type, data, coef, output
+
+
+@pytest.mark.parametrize('option', ['sample_weight', 'class_weight',
+                                    'balanced', 'no_weight'])
+def test_logistic_regression_weighting(regression_dataset, option):
+    regression_type, data, coef, output = regression_dataset
+
+    if option == 'sample_weight':
+        n_samples = data.shape[0]
+        sample_weight = np.abs(np.random.rand(n_samples))
+
+        culog = cuLog(fit_intercept=False)
+        culog.fit(data, output, sample_weight=sample_weight)
+
+        sklog = skLog(fit_intercept=False)
+        sklog.fit(data, output, sample_weight=sample_weight)
+    elif option == 'class_weight':
+        class_weight = np.random.rand(2)
```

Comment:
It would be good to print the random values when a test fails to help reproducibility by using a temporary varaible for the assert statements, for example https://github.com/rapidsai/cuml/blob/bacb05e276c960c02b3d59934bbc48a47f76a17b/python/cuml/test/experimental/test_explainer_kernel_shap.py#L66